### PR TITLE
Add 4 new command line options:

### PR DESCRIPTION
**Pull request by jcoleman1969**
-scale=<pct> to set pdfSettings.scale_factor
-delay=<ms> to delay after page load before rendering pdf (in my case allowed D3 charts to finish animating before creating the PDF)
-viewwidth=<px> to set viewport width (in my case setting the width to 1600 fixed a table that was trying to size to 100% and getting cut off at 128px wide, which is the default)
-viewheight=<px> added for parity, could be useful if element was specified as height=100%

### CODE REVIEW

**Comment by anko** [comment]
Nice!

I think `-delay` is fragile, because the necessary timeout differs depending on host machine specs, system load, input size, etc. In long-term, I'd want a callback for the page to fire when ready. For one-shots, `-delay` is useful as-is.

**Comment by jcoleman1969** [comment]
I just noticed that -delay doesn't seem to work consistently in the release build. I'm not sure why it's different than the debug build, but maybe just doing a sleep in Client::OnLoadEnd() isn't allowing the message loop to run?  I was thinking it may be better to set a flag in OnLoadEnd() that the load is finished and then call m_jobManager->Process() from elsewhere after the timeout has expired.  Do you have any ideas on where we could call it from?  Thanks!

**Comment by jcoleman1969** [comment]
I changed to using CefPostDelayedTask() instead of sleeping, and now --delay works reliably on both builds.

File: src/Client.cpp [hunk 1]
```diff
@@ -308,4 +322,16 @@ void Client::OnRenderProcessTerminated(
     DLOG(INFO) << "Client::OnRenderProcessTerminated";
 }
 
+void Client::SetViewWidth(int viewWidth)
+{
+   RenderHandler *renderHandler = (RenderHandler*)(m_renderHandler.get());
```
Comment (spajak) [inline src/Client.cpp]: Use ``CefRefPtr`` instead

File: src/main.cpp [hunk 2]
```diff
@@ -132,6 +136,22 @@ int runJob(CefRefPtr<cefpdf::Client> app, CefRefPtr<CefCommandLine> commandLine)
             job->SetBackgrounds();
         }
 
+        if (commandLine->HasSwitch("scale")) {
+            job->SetScale(atoi(commandLine->GetSwitchValue("scale").ToString().c_str()));
```
Comment (spajak) [inline src/main.cpp]: Use ``std`` namespace

File: src/main.cpp [hunk 3]
```diff
@@ -43,6 +43,10 @@ void printHelp(std::string name)
     std::cout << "                   If omitted some default margin is applied." << std::endl;
     std::cout << "  --javascript     Enable JavaScript." << std::endl;
     std::cout << "  --backgrounds    Print with backgrounds. Default is without." << std::endl;
+    std::cout << "  --scale=<%>      Scale the output. Default is 100." << std::endl;
+    std::cout << "  --delay=<ms>     Wait after page load before creating PDF. Default is 0." << std::endl;
+    std::cout << "  --viewwidth=<px> Width of viewport. Default is 128." << std::endl;
+    std::cout << "  --viewheight=<px> Height of viewport. Default is 128." << std::endl;
```
Comment (spajak) [inline src/main.cpp:49]: Align description texts

**Comment by jcoleman1969** [comment]
The CefRefPtr is declared as CefRefPtr<CefRenderHandler>, not to our
derived CefRefPtr<RenderHandler>, so I either have to downcast it, or I
need to change the declaration to CefRefPtr<RenderHandler> in Client.h.
Which would you prefer?

**Comment by spajak** [comment]
Yes. Change declaration. It should not break anything

**Comment by spajak** [comment]
Thanks. You could also update README if you can

**Pull request merged**